### DISCUSSION TITLE
How to get more FPS when using PyTorch Hub.

### DISCUSSION THREAD

**Discussion by shershunov** [ultralytics/yolov5#4004] [Q&A] [answered]
Without changing the number of parameters in the model.

**Comment by glenn-jocher** [comment, marked as answer]
@EkelviNistars 👋 Hello! Thanks for asking about **improving inference speed**. YOLOv5 🚀 can be run on CPU and GPU. When running inference on GPU the steps you can take to speed up inference are:

- Increase `--batch-size` for PyTorch Hub models
- Reduce `--img-size`
- Reduce model size, i.e. from YOLOv5x -> YOLOv5l -> YOLOv5m -> YOLOv5s
- Upgrade your hardware to a faster GPU

**Reply by shershunov** [reply]
How to set the batch-size parameter in PyTorch

**Reply by glenn-jocher** [reply]
```python
results = model([im1, im2, im3, etc...])
```
